### PR TITLE
Add Support for Intel with template functions

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: libdevcore/Base64.h [hunk 1]
```diff
@@ -51,6 +51,21 @@ template <size_t N> inline std::string toBase36(FixedHash<N> const& _h)
...
 
+#ifdef __INTEL_COMPILER
+inline std::string toBase36(FixedHash<Address::size> const& _h)
+{
+	static char const* c_alphabet = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+	typename FixedHash<Address::size>::Arith a = _h;
+	std::string ret;
+	for (; a > 0; a /= 36)
+	{
+		unsigned r = (a - a / 36 * 36).convert_to<unsigned>(); // boost's % is broken
```
Comment (ghost) [inline libdevcore/Base64.h]: This did not compile, reworking.

**Comment by ghost** [comment]
What is required to pull these changes?

The proposed changes will compile, and execute successfully.

Additional changes are ready for review regarding Xeon Phi, but those changes include this commit.

**Comment by bobsummerwill** [comment]
It's just time, @mancoast.
We have lots of tasks on the go.
@chriseth How do you want to proceed?   My vote would be to merge as-is, with a backlogged issue to revisit.

**Comment by chriseth** [comment]
@mancoast does this actually compile all of webthree-umbrella? If not, I would really be hesitant to merge this.

**Comment by ghost** [comment]
The code provides suitable binary path for libethereum with the following disabled:

```

#Disabled for Build
SET(FATDB OFF CACHE BOOL "Disable for Build")
SET(GUI OFF CACHE BOOL "Disable for Build")
SET(TESTS OFF CACHE BOOL "Disable for Build")
SET(ETHASHCL OFF CACHE BOOL "Disable for Build")
SET(EVMJIT OFF CACHE BOOL "Disable for Build")
SET(SOLIDITY OFF CACHE BOOL "Disable for Build")

```

**Comment by ghost** [comment]
confirmed build using boost_1_60_0

**Pull request closed without merging**